### ISSUE TITLE
WindowDescriptor resizing doesn't change internal sizes

### ISSUE THREAD

**Issue by rostyslav52** [bevyengine/bevy#605]
I tried printing the size of window (or to be precise WindowDescriptor) and noticed that after manually resizing the window, the internal sizes (defined during call to `new`) stay the same.

What's more, I tried to create a system that would update them. I was hoping that I would be able to contribute to Bevy and make a PR. 

I tried :

```
#[derive(Default)]
pub struct WindowResizedState {
    event_reader: EventReader<WindowResized>,
}

pub fn update_window_size(
    mut state: Local<WindowResizedState>,
    window_resized_events: Res<Events<WindowResized>>,
    mut window: ResMut<WindowDescriptor>
) {
    match state
        .event_reader
        .iter(&window_resized_events)
        .next() {
            Some(event) => {
                println!("Event size  : {} {}", event.width, event.height);
                window.width = event.width as u32;
                window.height = event.height as u32;
            }
            None => {}
        }
}
```
But it didn't work. I tried debugging a little and it seems that WindowResized events do not get fired when windows are manually resized. 

I would be happy to contribute to Bevy to modify this, but I couldn't find any source except for 
https://docs.rs/crate/bevy_window/0.2.1/source/src/

Or is it indeed all and there was nothing developed for window resizing?

P.S. By the way, WindowDescriptor is a Resource in my project, but can it be an Entity in theory?

**Comment by visualcode-t** [comment]
There is a WindowResized event in the src:
https://docs.rs/crate/bevy_window/0.2.1/source/src/event.rs

As far as resizing itself, I'm not sure.

**Comment by rostyslav52** [comment]
> There is a WindowResized event in the src:
> https://docs.rs/crate/bevy_window/0.2.1/source/src/event.rs

Yes, I am aware. This is the same link I provided in my post. I tried using this event in my code (sample in the original post), but it is never detected. 
It would be great to see code that is responsible for firing and catching WindowResized event :/

**Comment by visualcode-t** [comment]
A full working example would look like this:

```
use bevy::prelude::*;
use bevy::window::{WindowResized};

fn main() {
    App::build()
        .add_default_plugins()
        .add_system(window_resize.system())
        .run();
}

fn window_resize(resize_event: Res<Events<WindowResized>>) {
    let mut event_reader = resize_event.get_reader();
    for event in event_reader.iter(&resize_event) {
        println!("Info: {:?}",event);
    }
}
```

**Comment by rostyslav52** [comment]
Thanks, @visualcode-t 
Sorry for not answering for so long (October was a crazy month for me).

Indeed, you are completely right ! 

I will only add a small precision in case someone else ends up here : 

`use bevy::prelude::*;
use bevy::window::{WindowResized};
use std::convert::TryInto;

fn main() {
    App::build()
        .add_default_plugins()
        .add_resource(WindowDescriptor {
            title: "Toto".to_string(),
            width: 900,
            height: 600,
            vsync: true,
            resizable: true,
            //mode: WindowMode::Fullscreen { use_size: false },
            ..Default::default()
        })
        .add_system(window_resize.system())
        .run();
}

fn window_resize(resize_event: Res<Events<WindowResized>>,
                             mut window: ResMut<WindowDescriptor>) {
    let mut event_reader = resize_event.get_reader();
    for event in event_reader.iter(&resize_event) {
        window.width = event.width.try_into().unwrap();
        window.height = event.height.try_into().unwrap();
    }
}`

Basically, I just reused the code you provided, but added the **window.width = event.width** stuff.

**Comment by rostyslav52** [comment]
As @visualcode-t showed, Bevy already provides a simple solution to keep real window sizes synchronized after resizing. 
Maybe, it would be great to include something like this in a default plugin one day, but it I am perfectly happy with the current solution ;)
